e2e/_suites/metricbeat/metricbeat_test.go

```diff
@@ -403,12 +398,25 @@ func (mts *MetricbeatTestSuite) thereAreEventsInTheIndex() error {
 		},
 	}
 
-	result, err := e2e.RetrySearch(mts.getIndexName(), esQuery, queryMaxAttempts, queryRetryTimeout)
+	minimumHitsCount := 5
+	maxTimeout := time.Duration(queryRetryTimeout) * time.Minute
+
+	result, err := e2e.WaitForNumberOfHits(mts.getIndexName(), esQuery, minimumHitsCount, maxTimeout)
 	if err != nil {
 		return err
 	}
 
-	return e2e.AssertHitsArePresent(result, mts.Query)
+	err = e2e.AssertHitsArePresent(result)
```

Comment:
updating Metricbeats test at same time - you're doing good things here!  Very Elastic of you.